cpp/unreal_plugins/UrdfBot/Source/UrdfBot/UrdfJointComponent.cpp

```diff
@@ -48,56 +52,124 @@ void UUrdfJointComponent::initializeComponent(UrdfJointDesc* joint_desc, UUrdfLi
             ConstraintInstance.SetAngularDriveParams(spring, damping, force_limit);
             break;
         }
-    case UrdfJointType::Continuous:
-        {
-            float spring = FMath::RadiansToDegrees(joint_desc->damping_) * m_to_cm * 10.0f;
-            float damping = FMath::RadiansToDegrees(joint_desc->damping_) * m_to_cm;
-            float force_limit = FMath::RadiansToDegrees(joint_desc->effort_) * m_to_cm * 10.0f;
+    case UrdfJointType::Continuous: {
+        float spring = FMath::RadiansToDegrees(joint_desc->spring_) * m_to_cm;
+        float damping = FMath::RadiansToDegrees(joint_desc->damping_) * m_to_cm;
+        float force_limit = FMath::RadiansToDegrees(joint_desc->effort_) * m_to_cm;
 
-            // use twist degree for single angular freedom with best PhysX optimization
-            ConstraintInstance.SetAngularTwistLimit(EAngularConstraintMotion::ACM_Free, 0.0f);
-            ConstraintInstance.SetOrientationDriveTwistAndSwing(true, false);
-            ConstraintInstance.SetAngularVelocityDriveTwistAndSwing(true, false);
-            ConstraintInstance.SetAngularDriveParams(spring, damping, force_limit);
-            break;
+        // use twist degree for single angular freedom with best PhysX optimization
+        ConstraintInstance.SetAngularTwistLimit(EAngularConstraintMotion::ACM_Free, 0.0f);
+        ConstraintInstance.SetOrientationDriveTwistAndSwing(true, false);
+        ConstraintInstance.SetAngularVelocityDriveTwistAndSwing(true, false);
+        ConstraintInstance.SetAngularDriveParams(spring, damping, force_limit);
+        break;
+    }
+    case UrdfJointType::Prismatic: {
+        float linear_limit_size = (joint_desc->upper_ - joint_desc->lower_) * m_to_cm;
```

Comment:
Be consistent with your parentheses {} style. I like the style on line 67.

Reply:
fixed